### PR TITLE
completed task 8

### PR DESCRIPTION
**Pull request by nutmegvant**

### CODE REVIEW

File: model/app.model.js [hunk 1]
```diff
@@ -67,41 +67,66 @@ const selectAllCommentsForArticle = (id) => {
...
+        }    
+    )
+}
+
+function isEmpty(value) {
+    return (value == null || (typeof value === "string" && value.trim().length === 0));
+}
+
+const addNewVotes = (article_id, inc_votes) => {
+    if (isEmpty(inc_votes)){
```
Comment (nc-hannah-burks) [inline model/app.model.js:96]: If inc votes is missing, ideally we just want to return the article object unchanged, rather than throwing an error here :)

File: model/app.model.js [hunk 2]
```diff
@@ -67,41 +67,66 @@ const selectAllCommentsForArticle = (id) => {
...
-            } else {
-            return result.rows[0] }
-    }
-    )
-}
+            }
+            else {
+                const updatedVotes = result.rows[0].votes + inc_votes;
+                const updateParams = [updatedVotes, article_id]
+                return db.query(`UPDATE articles SET votes = $1 WHERE article_id = $2
```
Comment (nc-hannah-burks) [inline model/app.model.js:115]: This is fine, however, if you wanted to make it even simpler, rather than doing 
```
const updatedVotes = result.rows[0].votes + inc_votes;
                const updateParams = [updatedVotes, article_id]
```
You can instead, just add the votes via the sql like...
`return db.query(`UPDATE articles SET votes += $1 WHERE article_id = $2`
And then in your array, just put the amount of votes you are wanting to add and the id.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -66,6 +66,84 @@ describe("GET: /api/articles/:article_id", () => {
...
+        const votes = {inc_votes: 10}
+        return request(app)
+        .patch("/api/articles/abc")
+        .send(votes)
+        .expect(400)
+        .then((response) => {
+            expect(response.body.msg).toBe("Bad Request - type conversion issue");
+            });
+    });
+    test("PATCH: 400 lacking votes", () => {
```
Comment (nc-hannah-burks) [inline __tests__/app.test.js:127]: Ideally this would be Status 200, if inc_votes is missing, should not update article and return original single article object (this may be included as a 400 but that could make for a brittle test suite)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -66,6 +66,84 @@ describe("GET: /api/articles/:article_id", () => {
                 expect(response.body.msg).toBe("Article does not exist");
             });
     });
+    test('PATCH: 200 update votes on an article', () => {
+        const votes = {inc_votes: 10}
```
Comment (nc-hannah-burks) [inline __tests__/app.test.js:70]: You may also wish to test this with a negative number and see that it reduces the votes 😊


Overall, really good Nicola, just a couple minor tweaks
